neuralmonkey/experiment.py

```diff
@@ -0,0 +1,415 @@
+
+            training_loop(
+                tf_manager=self.model.tf_manager,
+                epochs=self.model.epochs,
+                trainer=self.model.trainer,
+                batch_size=self.model.batch_size,
+                log_directory=self.model.output,
+                evaluators=self.model.evaluation,
+                runners=self.model.runners,
+                train_dataset=self.model.train_dataset,
+                val_dataset=self.model.val_dataset,
+                test_datasets=self.model.test_datasets,
+                logging_period=self.model.logging_period,
+                validation_period=self.model.validation_period,
+                val_preview_input_series=self.model.val_preview_input_series,
+                val_preview_output_series=self.model.val_preview_output_series,
+                val_preview_num_examples=self.model.val_preview_num_examples,
+                postprocess=self.model.postprocess,
+                train_start_offset=self.model.train_start_offset,
+                runners_batch_size=self.model.runners_batch_size,
+                initial_variables=self.model.initial_variables)
+
+            self._vars_loaded = True
+
+    def load_variables(self, variable_files: List[str] = None) -> None:
+        if not self._model_built:
+            self.build_model()
+
+        if variable_files is None:
+            variable_files = [self.get_path("variables.data")]
+            log("Default variable file '{}' will be used for loading "
+                "variables.".format(variable_files[0]))
+
+        for vfile in variable_files:
+            if not os.path.exists("{}.index".format(vfile)):
+                raise RuntimeError(
+                    "Index file for var prefix {} does not exist"
+                    .format(vfile))
+
+        self.model.tf_manager.restore(variable_files)
+        self._vars_loaded = True
+
+    def run_model(self,
+                  dataset: Dataset,
+                  write_out: bool = False,
+                  batch_size: int = None,
+                  log_progress: int = 0) -> Tuple[List[ExecutionResult],
+                                                  Dict[str, List[Any]]]:
+        """Run the model on a given dataset.
+
+        Args:
+            dataset: The dataset on which the model will be executed.
+            write_out: Flag whether the outputs should be printed to a file
+                defined in the dataset object.
+            batch_size: size of the minibatch
+            log_progress: log progress every X seconds
+
+        Returns:
+            A list of `ExecutionResult`s and a dictionary of the output series.
+        """
+        if not self._model_built:
+            self.build_model()
+        if not self._vars_loaded:
+            self.load_variables()
+
+        with self.graph.as_default():
+            # TODO: check_dataset_and_coders(dataset, self.model.runners)
```

Comment:
to teď nejde takhle zavolat?

Reply:
Nějak si to stěžovalo, když tam nebyla series pro dekodér.